src/jit/codegenarm64.cpp

```diff
@@ -5059,9 +5059,104 @@ void CodeGen::genHWIntrinsicSimdBinaryOp(GenTreeHWIntrinsic* node)
     genProduceReg(node);
 }
 
+template <typename HWIntrinsicSwitchCaseBody>
+void CodeGen::genHWIntrinsicSwitchTable(regNumber                 swReg,
+                                        regNumber                 tmpReg,
+                                        int                       swMax,
+                                        HWIntrinsicSwitchCaseBody emitSwCase)
+{
+    assert(swMax > 0);
+    assert(swMax <= 256);
+
+    assert(genIsValidIntReg(tmpReg));
+    assert(genIsValidIntReg(swReg));
+
+    BasicBlock* labelFirst       = genCreateTempLabel();
+    BasicBlock* labelBreakTarget = genCreateTempLabel();
+
+    // Detect and throw out of range exception
+    getEmitter()->emitIns_R_I(INS_cmp, EA_4BYTE, swReg, swMax);
+
+    emitJumpKind jmpGEU = genJumpKindForOper(GT_GE, CK_UNSIGNED);
+    genJumpToThrowHlpBlk(jmpGEU, SCK_ARG_RNG_EXCPN);
+
+    // Calculate switch target
+    labelFirst->bbFlags |= BBF_JMP_TARGET;
+
+    // tmpReg = labelFirst
+    getEmitter()->emitIns_R_L(INS_adr, EA_PTRSIZE, labelFirst, tmpReg);
+
+    // tmpReg = labelFirst + swReg * 8
+    getEmitter()->emitIns_R_R_R_I(INS_add, EA_PTRSIZE, tmpReg, tmpReg, swReg, 3, INS_OPTS_LSL);
+
+    // br tmpReg
+    getEmitter()->emitIns_R(INS_br, EA_PTRSIZE, tmpReg);
+
+    genDefineTempLabel(labelFirst);
+    for (int i = 0; i < swMax; ++i)
+    {
+        emitSwCase(i);
+        inst_JMP(EJ_jmp, labelBreakTarget);
+    }
+    genDefineTempLabel(labelBreakTarget);
+}
+
 void CodeGen::genHWIntrinsicSimdExtractOp(GenTreeHWIntrinsic* node)
 {
-    NYI("HWIntrinsic form not implemented");
+    GenTree*  op1        = node->gtGetOp1();
+    GenTree*  op2        = node->gtGetOp2();
+    var_types simdType   = op1->TypeGet();
+    var_types targetType = node->TypeGet();
+    regNumber targetReg  = node->gtRegNum;
+
+    assert(targetReg != REG_NA);
+
+    genConsumeOperands(node);
+
+    regNumber op1Reg = op1->gtRegNum;
+
+    assert(genIsValidFloatReg(op1Reg));
+
+    emitAttr baseTypeSize = emitTypeSize(targetType);
+
+    int lanes = emitTypeSize(simdType) / baseTypeSize;
+
+    auto emitSwCase = [&](int lane) {
+        assert(lane >= 0);
```

Comment:
To me, `lane` is not very mnemonic, though others may disagree. Something like `caseImmediate` or `caseImm`?

Reply:
@CarolEidt 
The immediate is the vector element lane for this instruction.  

`caseImm` ... feels too generic.

`lane` and `lanes` could become:
+ `element` & `elements`
+ `vectorElement` & `vectorElements`
+ `vectorIndex` & `vectorLength`

Preference?

Reply:
Right - I hadn't really internalized that this is the specific case (not the general case as in `genHWIntrinsicSwitchTable()`). I think `element` and `elements` would be good.